app/Http/Controllers/Api/InventoryController1.php

```diff
@@ -0,0 +1,51 @@
+<?php namespace App\Http\Controllers;
+
+use App\Http\Requests;
+use App\Http\Controllers\Controller;
+use App\Item;
+
+use Illuminate\Support\Facades\Response;
+use Illuminate\Support\Facades\Input;
+
+class InventoryController1 extends Controller {
+
+	/**
+	 * Display a listing of the resource.
+	 *
+	 * @param  string $query
+	 * @return json all or queried $items
+	 */
+	public function index($query)
+	{
+		$items = Item::all();
+
+		if(!is_null($query)) {
+			$items = $items->find($query);
+		}
+
+		if(is_null($items)) {
+			return Response::json("Not Found", 404);
+		}
+		return $items;
+	}
+
+
+	/**
+	 * Store a newly created resource in storage.
+	 *
+	 * @param  none
+	 * @return Json input $item
+	 */
+	public function store()
+	{
+		$input = Input::all();
+		if(is_null($input)) {
+			return Response::json("No Input", 406);
```

Comment:
This is indeed where we would put validation, but we probably want to do more than test whether `$input` is null.

Also, we want to use HTTP 400 (bad request) if the input is invalid. 406 has to do with acceptable content types. We won't have to worry about that here.